source/view.js

```diff
@@ -4119,12 +4133,38 @@ view.Tensor = class {
     }
 
     get metrics() {
-        const metrics = Array.from(this._tensor.metrics || []);
-        const keys = new Set(metrics.map((metrics) => metrics.name));
-        if (!keys.has('sparisity')) {
-            // metrics.push(new view.Argument('sparisity', 0, 'float32'));
+        if (this._metrics === null) {
+            const value = this.value;
+
+            const metrics = Array.from(this._tensor.metrics || []);
+            const keys = new Set(metrics.map((metrics) => metrics.name));
+            if (!keys.has('sparsity')) {
+                let num_zeros = 0;
+                let num_parameters = 0;
+                const stack = [value];
+                while (stack.length > 0) {
+                    const val = stack.pop();
+                    if (Array.isArray(val)) {
+                        for (const element of val) {
+                            stack.push(element);
+                        }
+                    } else {
+                        num_zeros += Number(val === 0);
+                        num_parameters += 1;
+                    }
+                }
+
+                if (num_parameters > 0) {
+                    metrics.push(new view.Argument('sparsity', num_zeros / num_parameters, 'float32', 'percentage'));
+                } else {
+                    metrics.push(new view.Argument('sparsity', 0, 'float32', 'percentage'));
```

Comment:
Is percentage more of a special case or are there other such metric styles that might be needed?

Reply:
Percentage would be used by other metrics such as operation sparsity and four block sparsity

Reply:
Is percentage a special case of a `float32` value. `Argument` is a data concept and implemented pretty consistently. Introducing `style` would mix presentation with data concepts and this would potentially have to extend across all API layers. The alternative is handling it as a special `type`.

Reply:
> Argument is a data concept and implemented pretty consistently

You're correct, implementing the member `style` was a mistake. 

I initially avoided adding a special case of `type` since it's another case which has to be handled by functions which interpret `Argument`, for example #1241. However, on second thought, the fact that an `Argument` is a percentage also affects how it's interpreted.

In any case, I agree, this is probably best implemented as a special `float32-perc` style.